app/src/main/java/org/fossasia/openevent/app/common/utils/core/DateUtils.java

```diff
@@ -78,17 +86,11 @@ public static String formatDate(@NonNull String format, @NonNull String isoDateS
 
     @NonNull
     public static String formatDateWithDefault(@NonNull String format, @NonNull String isoString, @NonNull String defaultString) {
-        try {
+        getDate(isoString);
+        if (!exception)
```

Comment:
This will return default string after any parsing error if getDate is not called and as they are not part of same function, it is impossible to debug the implications of state change. Furthermore, I have not even considered synchronization issues in a parallel thread. This solution is dead ringer for countless more bugs than it solves

Recognize the error and solve it rather than patching unneeded things :smile:

Reply:
`return ZonedDateTime.parse(isoDateString).withZoneSameInstant(getZoneId());`
This line causes a DateTimeParse exception in getDate
So the only thing that I could think of was adding a try catch block.
If I don't return current date when a string is invalid, what should I return?
Other than the code written in getDate, the rest of the code is a workaround to pass the unit test

Reply:
Tell me this. If you ask someone time and their clock is broken, should they tell you that their clock is broken or just tell you any time. The method **should** throw exception if the date is in wrong format and not return current time

The problem is that the date should never be in wrong format, so try to find out on what cases it is throwing exception and remove those cases from happening

Reply:
The date will be only be in wrong format when we manually change it in the editText(like it is being done in the gif)
Can making the editText non-editable be a solution?

Reply:
Or you could try catch but in the calling method and not in the util method. Util method should always throw error in wrong format